app/status.go

```diff
@@ -295,6 +295,45 @@ func (a *App) SetStatusAwayIfNeeded(userId string, manual bool) {
 	})
 }
 
+func (a *App) SetStatusDndIfNeeded(userId string, manual bool) {
+	if !*a.Config().ServiceSettings.EnableUserStatuses {
+		return
+	}
+
+	status, err := a.GetStatus(userId)
+
+	if err != nil {
+		status = &model.Status{UserId: userId, Status: model.STATUS_OFFLINE, Manual: manual, LastActivityAt: 0, ActiveChannel: ""}
+	}
+
+	if !manual && status.Manual {
+		return // manually set status always overrides non-manual one
+	}
+
+	if !manual {
+		if status.Status == model.STATUS_DND {
+			return
+		}
+	}
```

Comment:
Per above comment, I don't think you need to check `manual` since this status can only be set manually

Reply:
removed